src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,1443 @@
+            }
+            tmpAlkylSFBondIndex++;
+        }
+        ArrayList<IBond> tmpRemovedNull = new ArrayList<IBond>();
+        for (IBond tmpBond: tmpBondArray)
+            if (tmpBond != null) {
+                tmpRemovedNull.add(tmpBond);
+            }
+        return tmpRemovedNull.toArray(new IBond[0]);
+    }
+    /**
+     * Protected method for detecting and marking tertiary or quaternary carbon atoms and their surrounding neighbor atoms and bonds.
+     *
+     * @param aMolecularArraysInstance MolecularArrays instance for data transfer between methods
+     * @param anAtomArray Given array with atoms of molecule to be fragmented
+     * @param aBondArray Given array with bonds of molecule to be fragmented
+     */
+    protected void markNeighborAtomsAndBonds(MolecularArrays aMolecularArraysInstance,IAtom[] anAtomArray, IBond[] aBondArray) {
+        Objects.requireNonNull(anAtomArray, "Given atom array is null.");
+        Objects.requireNonNull(aBondArray,"Given bond array is null");
+        //set general atom and specific bond properties
+        for (int tmpAtomIndex = 0; tmpAtomIndex < anAtomArray.length; tmpAtomIndex++) {
+            IAtom tmpAtom = anAtomArray[tmpAtomIndex];
+            if (tmpAtom != null) {
+                //marking of tertiary or quaternary carbons together with their neighbor atoms
+                if (tmpAtom.getBondCount() == 3 && tmpAtom.getMaxBondOrder() == IBond.Order.SINGLE) {
+                    tmpAtom.setProperty(AlkylStructureFragmenter.INTERNAL_ASF_TERTIARY_CARBON_PROPERTY_KEY, true);
+                    for (IBond tmpBond: tmpAtom.bonds()) {
+                        int tmpBondIndex = tmpBond.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_BOND_INDEX_PROPERTY_KEY);
+                        int tmpBondBeginIndex = tmpBond.getBegin().getProperty(AlkylStructureFragmenter.INTERNAL_ASF_ATOM_INDEX_PROPERTY_KEY);
+                        int tmpBondEndIndex = tmpBond.getEnd().getProperty(AlkylStructureFragmenter.INTERNAL_ASF_ATOM_INDEX_PROPERTY_KEY);
+                        if (tmpBond.getBegin() == tmpAtom) {
+                            anAtomArray[tmpBondEndIndex].setProperty(AlkylStructureFragmenter.INTERNAL_ASF_NEIGHBOR_MARKER_KEY, true);
+                            aBondArray[tmpBondIndex].setProperty(AlkylStructureFragmenter.INTERNAL_ASF_NEIGHBOR_MARKER_KEY, true);
+                        } else if (tmpBond.getEnd() == tmpAtom) {
+                            anAtomArray[tmpBondBeginIndex].setProperty(AlkylStructureFragmenter.INTERNAL_ASF_NEIGHBOR_MARKER_KEY, true);
+                            aBondArray[tmpBondIndex].setProperty(AlkylStructureFragmenter.INTERNAL_ASF_NEIGHBOR_MARKER_KEY, true);
+                        }
+                    }
+                } else if (tmpAtom.getBondCount() == 4 && tmpAtom.getMaxBondOrder() == IBond.Order.SINGLE) {
+                    tmpAtom.setProperty(AlkylStructureFragmenter.INTERNAL_ASF_QUATERNARY_CARBON_PROPERTY_KEY, true);
```

Comment:
Only this one line is different from the if case above, so duplication can be reduced here.